go/runtime/bundle/manifest.go

```diff
@@ -148,6 +151,38 @@ func (c ComponentID) String() string {
 	return string(c.Kind) + " (" + c.Name + ")"
 }
 
+// MarshalText serializes the component identifier into text form.
+func (c ComponentID) MarshalText() ([]byte, error) {
+	return []byte(string(c.Kind) + kindNameSeparator + c.Name), nil
+}
+
+// UnmarshalText deserializes the component identifier from text form.
+func (c *ComponentID) UnmarshalText(text []byte) error {
+	atoms := strings.SplitN(string(text), kindNameSeparator, 2)
+
+	var kind ComponentKind
+	switch atoms[0] {
+	case string(ComponentRONL):
+		kind = ComponentRONL
+	case string(ComponentROFL):
+		kind = ComponentROFL
+	default:
+		return fmt.Errorf("malformed component kind: %s", atoms[0])
+	}
+
+	if kind == ComponentRONL && len(atoms) > 1 && len(atoms[1]) > 0 {
+		return fmt.Errorf("RONL component must have an empty name, got: %s", atoms[1])
```

Comment:
```suggestion
	if c.Kind == ComponentRONL && len(c.Name) > 0 {
		return fmt.Errorf("RONL component must have an empty name, got: %s", c.Name)
	}
```

Reply:
Ignore, as you probably don't want to change `c` if you return an error.

Reply:
Yeah exactly.